subprojects/docs/src/samples/jvm-multi-project-with-code-coverage/groovy/build.gradle

```diff
@@ -0,0 +1,58 @@
+plugins {
+    id 'java'
+    id 'jacoco'
+}
+
+allprojects {
+    version = '1.0.2'
+    group = 'org.gradle.sample'
+
+    repositories {
+        jcenter()
+    }
+}
+
+subprojects {
+    tasks.withType(Test).configureEach {
+        useJUnitPlatform()
+    }
+}
+
+// tag::coverageTask[]
+// task to gather code coverage from multiple subprojects
+// NOTE: the report task does *not* depend on the `test` task by default. Meaning you have to ensure
+// that `test` (or other tasks generating code coverage) run before generating the report.
+// You can achieve this by calling the `check` lifecycle task manually
+// $ ./gradlew test codeCoverageReport
+tasks.register("codeCoverageReport", JacocoReport) {
+    // If a subproject applies the 'jacoco' plugin, add the result it to the report
+    subprojects { subproject ->
+        subproject.plugins.withType(JacocoPlugin).configureEach {
+            subproject.tasks.matching({ t -> t.extensions.findByType(JacocoTaskExtension) }).configureEach { testTask ->
+                sourceSets subproject.sourceSets.main
+                executionData(testTask)
+            }
+
+            // Alternatively you can declare a dependency so that the `codeCoverageReport` depends on the `test` tasks.
```

Comment:
:o: It's probably not clear that "Alternatively" here now relates to the text block further up. Instead you could give the full context again, like:

> To automatically run `test` overtime `./gradlew codeCoverageReport` is called...

And the you can remove this sentence below:
"This inevitably enforces that running the `codeCoverageTask` runs the test tasks as well."